tools/packaging/static-build/kernel/build.sh

```diff
@@ -24,50 +28,66 @@ container_image="${KERNEL_CONTAINER_BUILDER:-$(get_kernel_image_name)}"
 MEASURED_ROOTFS=${MEASURED_ROOTFS:-no}
 kernel_builder_args="-a ${ARCH} $*"
 
-if [ "${MEASURED_ROOTFS}" == "yes" ]; then
+if [[ "${MEASURED_ROOTFS}" == "yes" ]]; then
 	info "build initramfs for cc kernel"
 	"${initramfs_builder}"
 	# Turn on the flag to build the kernel with support to
 	# measured rootfs.
 	kernel_builder_args+=" -m"
 fi
 
-if [ "${CROSS_BUILD}" == "true" ]; then
+if [[ "${CROSS_BUILD}" == "true" ]]; then
        container_image="${container_image}-${ARCH}-cross-build"
        # Need to build a s390x image due to an issue at
        # https://github.com/kata-containers/kata-containers/pull/6586#issuecomment-1603189242
-       if [ ${ARCH} == "s390x" ]; then
+       if [[ ${ARCH} == "s390x" ]]; then
                BUILDX="buildx"
                PLATFORM="--platform=linux/s390x"
        fi
 fi
 
-docker pull ${container_image} || \
-	(docker ${BUILDX} build ${PLATFORM} \
-	--build-arg ARCH=${ARCH} -t "${container_image}" "${script_dir}" && \
-	 # No-op unless PUSH_TO_REGISTRY is exported as "yes"
-	 push_to_registry "${container_image}")
+container_engine=${CONTAINER_ENGINE:-"docker"}
+container_build=${container_engine}
+
+if [[ -n "${BUILDX}" ]]; then
+	container_build+=" ${BUILDX}"
+fi
+
+container_build+=" build"
+
+if [[ -n "${PLATFORM}" ]]; then
+	container_build+=" ${PLATFORM}"
+fi
+
+container_build+=" --build-arg ARCH=${ARCH}"
```

Comment:
It also feels like this might end up being a common pattern we use elsewhere. It's not super readable though :(. Is there a requirement to support container engines other than docker or could a podman user just alias to help make this a bit simpler?

Reply:
That's why I put it like that to see if this is needed or not. We could even simplify that and make it more readable. The issue is that BUILDX and PLATFORM, if not set, will eval to `' '` and then docker gets confused 
```
docker ' ' build ' ' .... 
```

Reply:
rootfs.sh e.g. has a nicer way of doing this with `engine_args...` but then if not done right confused  docker, the argv/argc handling of docker is just horrible empty strings are interpreted as something important.

Reply:
Need to think about that and reiterate over this.

Reply:
Thanks. I don't think it's a hard blocker, it just took me a little while to read and understand what was going on. It is Friday afternoon for me, so that could be why!

Reply:
What was the original error? This is indeed much harder to read unfortunately

Reply:
> That's why I put it like that to see if this is needed or not. We could even simplify that and make it more readable. The issue is that BUILDX and PLATFORM, if not set, will eval to `' '` and then docker gets confused
> 
> ```
> docker ' ' build ' ' .... 
> ```

@sprt See above comment